05_fft.md

```diff
@@ -0,0 +1,239 @@
+$$
+
+## Fourier Transform
+By construction, the Fourier series we defined above is periodic on an interval
+of $2l$. Evaluating it outside $-l < x < l$, we get copies of the same function.
+
+![](./plots/fft_ex_fun2_period.png){ width=50% }
+
+## Fourier Transform
+What if we want to decompose a function that is not periodic?
+We can extend the domain to be arbitrarily large:
+$$
+l \rightarrow \infty \text{,} \quad \frac{n}{2 l} \rightarrow k \text{,}
+$$
+where $k$ is the continuous limit of $n$, the "frequency" of the sine function.
+The sum over $n$ becomes an integral over $k$, and the coefficients $c_n$, a
+continuous function of $k$
+$$
+  f(x) = \int_{-\infty}^{\infty} \frac{dk}{2 \pi}~e^{2 \pi i k x} F(k) \text{.}
+$$
+
+## Fourier Transform
+$F(k)$ is the Fourier transform of $f(x)$. Because it is common to take Fourier
+transforms of time-series data, we often refer to $f(x)$ as being in the "time
+domain" as $F(k)$ its representation in the "frequency domain". There is no
+fundamental difference between the two, the inverse transform has the same form
+and is lossless.
+$$
+  F(k) = \int_{-\infty}^{\infty} dx~e^{-2 \pi i k x} f(x) \text{,}
+$$
+$$
+  f(x) = \int_{-\infty}^{\infty} \frac{dk}{2 \pi}~e^{2 \pi i k x} F(k) \text{.}
+$$
+
+## Fourier Transform of a Real Function
+$$
+\begin{aligned}
+  F(-k) &= \int_{-\infty}^{\infty} dx~e^{-2 \pi i (-k) x} f(x) \\
+  &= \left[\int_{-\infty}^{\infty} dx~e^{-2 \pi i k x} f^*(x)\right]^* \\
+  &= \left[\int_{-\infty}^{\infty} dx~e^{-2 \pi i k x} f(x)\right]^* \\
+  &= F^*(k) \text{.}
+\end{aligned}
+$$
+The negative frequencies are redundant.
+
+## Fourier Transform of a Real Function
+$$
+  F(-k) = F^*(k)
+$$
+![](./plots/fft_ex_fun2_ft.png){ width=100% }
+
+## Convolution theorem
+**(skipping this)**
+
+## Sampling Theorem
+A measurement is composed of discrete samples of a continuous signal, separated in time by intervals $\Delta t$.
+The sampling theorem states that for a *bandlimited signal* (i.e. with power only in a finite frequency band, say $\Delta\nu$) there is a finite sampling rate beyond which *no additional information* is obtained about the underlying signal.
+It is called the Nyquist rate and is twice the bandwidth:
+$$
+  \frac{1}{\Delta t} \geq 2 \Delta \nu \text{.}
+$$
+If a signal is Nyquist sampled, the underlying continuous signal can be perfectly reconstructed (in the absence of noise).
+
+## Sampling Theorem Example
+The human ear can only hear frequencies below $\sim$ 20 kHz.
+A perceptible sound signal is effectively band-limited at 20 kHz.
+This is why high-fidelity recordings (e.g. CD quality) are sampled around 44 kHz.
+This is sufficient to reconstruct the original sound perfectly.
+
+## Aliasing
+What if the sampling rate is too low to capture the highest frequencies?
+$$
+  \Delta t \lt \frac{1}{2 \Delta\nu}
+$$
+It becomes impossible to distinguish between certain pairs of frequencies.
+
+![](./plots/fft_alias.png){ width=50% }
+
+## Aliasing in Ringmaps
```

Comment:
It's not clear that they now that ring map imaging is an FFT yet. You could maybe move this until the end.